detect-nuget-inspector/detect-nuget-inspector/DependencyResolution/Project/ProjectXmlResolver.cs

```diff
@@ -13,12 +13,23 @@ class ProjectXmlResolver : DependencyResolver
 
         private string ProjectPath;
         private NugetSearchService NugetSearchService;
+        private HashSet<PackageId> Packages;
+        private bool CheckVersionOverride;
 
         public ProjectXmlResolver(string projectPath, NugetSearchService nugetSearchService)
         {
             ProjectPath = projectPath;
             NugetSearchService = nugetSearchService;
         }
+        
+        public ProjectXmlResolver(string projectPath, NugetSearchService nugetSearchService, HashSet<PackageId> packages, bool checkVersionOverride)
```

Comment:
Add another constructor to extend support and pass along Package Versions which are parsed from Directory.Packages.props file. This will ensure we differentiate between Nuget Project Inspector and Nuget Solution Inspector.